docs/adrs/0262-custom-label-support.md

```diff
@@ -0,0 +1,43 @@
+# ADR 262: Custom Label Support
+**Date**: 2020-01-14
+
+**Status**: Proposed
+
+## Context
+
+In the case of a single repo having multiple self-hosted runners, actions do not currently support any way of selecting which runner or kind of runner a job should run on, other than by architecture or OS.
+
+Example use cases for this could be:
+
+* Your project involves some machine learning models. You want to add a second runner with a GPU for testing your models, and to have your model tests job use this runner, but other jobs to use the runner without a GPU. 
+* You want to use GitHub actions to deploy your app, and for security you use a self-hosted runner inside your isolated network. If you have a separate isolated network for each environment (e.g. testing, staging, production), you would need to run your deploy job on a different runner depending on which environment you are deploying to.
+* Any other situation where you would like to select which self-hosted runner to use for a job based on the capabilities or characteristics of the runner.
+
+GitHub actions already support selecting a job runner based on labels, and automatically give runners labels based on OS and arch, so a natural extension would be to allow operators of self-hosted runners to add their own labels, so that users can select job runners based on any operator-defined characteristic.
+
+See Issue: https://github.com/actions/runner/issues/262
+
+## Decision
+
+This ADR proposes that we add a `--label` option to `config.sh`, which could be used to add custom additional labels to the configured runner.
+
+For example, to add a single extra label the operator could run:
+```bash
+./config.sh --label my-extra-label
+```
+Which would add the label `my-extra-label` to the runner, and enable users to select the runner in their workflow using this label:
+```yaml
+runs-on: [self-hosted, my-extra-label]
+```
+
+To add multiple labels:
+```bash
+./config.sh --label my-extra-label --label yet-another-label
+```
+This would add both labels `my-extra-label` and `yet-another-label`.
+
```

Comment:
Address remove label.  I think it should only be a web UI action since config is typically a one time operation.  That does leave a gap until we have web UI support.  In that gap, the (not ideal) workaround would be to unconfigure and reconfigure.  On configure, I think we should rewrite the set.  That means there's no cli arg to remove.

Reply:
Another way is to use file, user put labels into `.labels` file under runner root, the runner pick them up during runner config and session creation, so if user want to change labels after config, they just need to change the file and restart the runner process.

Reply:
That would get interesting depending on how the UI will work. I need to get some design progress on that. Hopefully by next week I can have something more fleshed out.

Reply:
I've added a line about removing labels: 
`It would not be possible to remove labels from an existing runner using `config.sh`, instead labels would have to be removed using the GitHub UI.
`